File: library/src/main/java/io/oddworks/device/exception/BadResponseCodeException.java
package io.oddworks.device.exception;

/**
 * Response code returned from server was not what was expected
 */
public class BadResponseCodeException extends OddRequestException {
    private final int code;

    public BadResponseCodeException(int code) {
        this.code = code;
    }

    public BadResponseCodeException(String detailMessage, int code) {
        super(detailMessage);
        this.code = code;
    }

    public BadResponseCodeException(String detailMessage, Throwable throwable, int code) {
        super(detailMessage, throwable);
        this.code = code;
    }

    public BadResponseCodeException(Throwable throwable, int code) {
        super(throwable);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/exception/BadResponseCodeException.kt
package io.oddworks.device.exception

import io.oddworks.device.model.OddError
import java.util.*

/**
 * Response code returned from server was not what was expected
 */
class BadResponseCodeException : OddRequestException {
    val code: Int
    val oddErrors: LinkedHashSet<OddError>

    constructor(code: Int) {
        this.code = code
        this.oddErrors = linkedSetOf()
    }

    constructor(code: Int, oddErrors: LinkedHashSet<OddError>) {
        this.code = code
        this.oddErrors = oddErrors
    }
}
